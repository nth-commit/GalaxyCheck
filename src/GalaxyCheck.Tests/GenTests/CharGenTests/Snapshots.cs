using GalaxyCheck;
using Snapshooter;
using Snapshooter.Xunit;
using System.Linq;
using Xunit;

namespace Tests.V2.GenTests.CharGenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 3);

            foreach (var seed in seeds)
            {
                var sample = Gen.Char().Select(c => char.IsControl(c) ? "<control>" : c.ToString()).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 3);

            foreach (var seed in seeds)
            {
                var sample = Gen.Char()
                    .Select(c => char.IsControl(c) ? "<control>" : c.ToString())
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x);

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
