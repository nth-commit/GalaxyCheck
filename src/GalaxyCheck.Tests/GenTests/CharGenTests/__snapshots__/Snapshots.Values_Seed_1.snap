[
  "a",
  "a",
  "a",
  "b",
  "a",
  "a",
  "b",
  "b",
  "b",
  "a",
  "a",
  "b",
  "a",
  "b",
  "b",
  "b",
  "b",
  "a",
  "a",
  "b",
  "d",
  "b",
  "a",
  "e",
  "f",
  "b",
  "f",
  "c",
  "b",
  "d",
  "d",
  "d",
  "c",
  "a",
  "c",
  "b",
  "f",
  "d",
  "a",
  "a",
  "g",
  "f",
  "j",
  "c",
  "g",
  "d",
  "e",
  "e",
  "f",
  "d",
  "h",
  "d",
  "i",
  "c",
  "k",
  "i",
  "e",
  "c",
  "i",
  "d",
  "&",
  "B",
  "<control>",
  "r",
  ")",
  "<control>",
  "1",
  "a",
  "-",
  "Z",
  "<control>",
  "Z",
  "C",
  "Q",
  "t",
  "E",
  "F",
  ":",
  "V",
  "<control>",
  "Z",
  "<control>",
  "O",
  "F",
  "<control>",
  "<control>",
  "<control>",
  "¦",
  "<control>",
  "<control>",
  "<control>",
  "<control>",
  "<control>",
  "-",
  "<control>",
  "È",
  "c",
  "<control>",
  "^",
  "¼"
]
